src/cryptsetup/cryptsetup-fido2.c

```diff
@@ -243,20 +246,36 @@ int find_fido2_auto_data(
                         SET_FLAG(required, FIDO2ENROLL_UV, json_variant_boolean(w));
                 } else
                         required |= FIDO2ENROLL_UV_OMIT; /* compat with 248 */
+
+                r = acquire_fido2_key(
+                                name,
+                                friendly_name,
+                                fido2_device,
+                                rp,
+                                cid, cid_size,
+                                key_file, key_file_size, key_file_offset,
+                                salt, salt_size,
+                                until,
+                                headless,
+                                required,
+                                ret_decrypted_key, ret_decrypted_key_size,
+                                ask_password_flags);
+
+                if (r == 0)
+                        break;
         }
 
         if (!cid)
                 return log_error_errno(SYNTHETIC_ERRNO(ENXIO),
                                        "No valid FIDO2 token data found.");
 
-        log_info("Automatically discovered security FIDO2 token unlocks volume.");
+        if (r < 0) {
+                if (r == -EAGAIN) /* fido2 device does not exist, or UV is blocked (reinsertion needed) */
+                        return log_error_errno(r,
+                                               "FIDO2 token not available");
```

Comment:
should not be an error? i.e. not log_error_errno()...

i'd write this like this:

```c
if (r == -EAGAIN)
        …
if (r < 0)
        …
```

in the interest of keeping indentation level low/reduce nesting